src/Common/src/Interop/Windows/kernel32/Interop.GetConsoleTitle.cs

```diff
@@ -5,68 +5,54 @@
 using System.Runtime.InteropServices;
+using System;
 
 internal partial class Interop
 {
     internal partial class Kernel32
     {
-        [DllImport(Libraries.Kernel32, CharSet = CharSet.Unicode, SetLastError = true, EntryPoint = "GetConsoleTitleW")]
-        private static extern int GetConsoleTitle([Out]StringBuilder title, int nSize);
+        [DllImport(Libraries.Kernel32, CharSet = CharSet.Unicode, SetLastError = true, ExactSpelling = true)]
+        private static extern uint GetConsoleTitleW(ref char title, uint nSize);
 
-        private const int InitialBufferSizeInChars = 256;
-
-        // Although msdn states that the max allowed limit is 65K,
-        // desktop limits this to 24500 as buffer sizes greater than it
-        // throw.
-        private const int MaxAllowedBufferSizeInChars = 24500;
-
-        // GetConsoleTitle sometimes interprets the second parameter (nSize) as number of bytes and sometimes as the number of chars.
-        // Instead of doing complicated and dangerous logic to determine if this may or may not occur,
-        // we simply assume the worst and reserve a bigger buffer. This way we may use a bit more memory,
-        // but we will always be safe.
-        private static int CharCountToByteCount(int numChars)
+        internal static string GetConsoleTitle(out int error)
         {
-            return numChars * 2;
-        }
+            error = Errors.ERROR_SUCCESS;
 
-        // 1. We first try to call the GetConsoleTitle with an InitialBufferSizeInChars of 256.
-        // 2. Then based on the return value either increase the capacity or return failure.
-        internal static int GetConsoleTitle(out string title, out int titleLength)
-        {
-            int lastError = 0;
-            StringBuilder sb = new StringBuilder(CharCountToByteCount(InitialBufferSizeInChars + 1));
-            int len = GetConsoleTitle(sb, sb.Capacity);
+            Span<char> initialBuffer = stackalloc char[256];
+            ValueStringBuilder builder = new ValueStringBuilder(initialBuffer);
 
-            if (len <= 0)
+            do
             {
-                lastError = Marshal.GetLastWin32Error();
-            }
-            else if (len > MaxAllowedBufferSizeInChars)
-            {
-                // Title is greater than the allowed buffer so we do not read the title and only pass the length to the caller.
-                title = string.Empty;
-                titleLength = len;
-                return 0;
-            }
-            else
-            {
-                if (len > InitialBufferSizeInChars)
+                // The documentation asserts that the console's title is stored in a shared 64KB buffer.
+                // The magic number that used to exist here (24500), is likely related to that.
+                // A full UNICODE_STRING is 32K chars...
+                Debug.Assert(builder.Capacity <= short.MaxValue, "shouldn't be possible to grow beyond UNICODE_STRING size");
```

Comment:
- If the console length is 24500, I think this will grow to `short.MaxValue + 1`.
- Capacity is actually what the array pool gave you, it can be much bigger than what you asked for.

Reply:
Good catch, I'll update to check result. Result shouldn't be short.MaxValue + 1.